app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -204,15 +206,25 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
             requireContext(),
             requireContext().settings(),
             interactor,
-            historyStorageProvider(),
             isPrivate,
             binding.toolbar,
-            requireComponents.core.engine,
             fromHomeFragment
         ).also {
             inlineAutocompleteEditText = it.view.findViewById(R.id.mozac_browser_toolbar_edit_url_view)
         }
 
+        val engineForSpeculativeConnects = if (!isPrivate) requireComponents.core.engine else null
```

Comment:
Let's move this into the if block since the variable is only needed inside.

Reply:
Good idea.  I'll update.  Thanks